packages/api/amplify_api/lib/src/util.dart

```diff
@@ -30,3 +30,21 @@ Map<String, String>? addContentTypeToHeaders(
   modifiedHeaders.putIfAbsent(AWSHeaders.contentType, () => contentType);
   return modifiedHeaders;
 }
+
+/// Grabs errors from GraphQL Response. Is used in method channel and Dart first code.
+/// TODO: Move to Dart first code when method channel GraphQL implementation is removed.
+@internal
+List<GraphQLResponseError> deserializeGraphQLResponseErrors(
+  Map<String, dynamic> response,
+) {
+  final errors = response['errors'] as List?;
+  if (errors == null || errors.isEmpty) {
+    return const [];
```

Comment:
should this return null and then the return type of this method would be `List<GraphQLResponseError>?`